### PR TITLE
Master account new api general ledger dmi

### PR DESCRIPTION
**Pull request by dmichiels**

### CODE REVIEW

File: addons/account/report/account_bank_reconciliation_report.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            'columns': [line.date, line.ref, amount],
+            'level': 3,
+        }
+
+    @api.model
+    def _lines(self):
+        lines = []
+        #Start amount
+        account_ids = list(set([self.env.context['journal_id'].default_debit_account_id.id, self.env.context['journal_id'].default_credit_account_id.id]))
+        lines_already_accounted = self.env['account.move.line'].search(['|', ('move_id.journal_id.type', '=', 'general'),
```
Comment (pvy-odoo) [inline addons/account/report/account_bank_reconciliation_report.py]: Maybe use query_get() ?

File: addons/account/views/report_financial.xml [hunk 2]
```diff
@@ -37,6 +37,16 @@
     <field name="url">/account/generic_tax_report/1</field>
     <field name="target">self</field>
 </record>
+<record model="ir.actions.act_url" id="action_account_report_bank_reconciliation">
+    <field name="name">Bank Reconciliation</field>
+    <field name="url">/account/bank_reconciliation/1</field>
+    <field name="target">self</field>
+</record>
+<record model="ir.actions.act_url" id="action_account_report_general_ledger">
+    <field name="name">Bank Reconciliation</field>
```
Comment (pvy-odoo) [inline addons/account/views/report_financial.xml]: Should be General Ledger

File: addons/account/views/report_configurator.xml [hunk 3]
```diff
@@ -30,7 +30,19 @@
                         </t>   
                     </select>
                 </div>
-                <div class="col-sm-3">
+		<div class="col-sm-3" t-if="report.get_report_type() == 'bank_reconciliation'">
```
Comment (pvy-odoo) [inline addons/account/views/report_configurator.xml]: Maybe we should create a new model that would be used to manage all those kinds of filters

File: addons/account/report/account_general_ledger.py [hunk 4]
```diff
@@ -1,274 +1,158 @@
...
-        del counterpart_res
+    def _lines(self, line_id = None):
+        lines = []
+        context = self.env.context
+        domain = [('date', '>=', context['date_from']), ('date', '<=', context['date_to'])]
+        if line_id:
+            domain.append(('account_id', '=', line_id))
+        if context['target_move'] == 'posted':
+            domain.append(('move_id.state', '=', 'posted'))
+        move_line_ids = self.env['account.move.line'].search(domain).sorted(key=lambda r: r.date)
```
Comment (pvy-odoo) [inline addons/account/report/account_general_ledger.py]: Again maybe use query_get() ?

File: addons/account/report/account_bank_reconciliation_report.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+
+
+class account_bank_reconciliation(models.AbstractModel):
+    _name = 'account.bank.reconciliation'
```
Comment (pvy-odoo) [inline addons/account/report/account_bank_reconciliation_report.py]: maybe add report in the model name

File: addons/account/account_move_line.py [hunk 6]
```diff
@@ -89,6 +89,17 @@ def _compute_cash_basis(self):
...
 
+    @api.one
+    @api.depends('move_id.line_id')
+    def _get_counterpart(self):
+        counterpart = set()
+        for line in self.move_id.line_id:
+            if (line.account_id.code != self.account_id.code):
+                counterpart.add(line.account_id.code)
+        if len(counterpart) > 2:
+            counterpart = counterpart[0:2] + ["..."]
```
Comment (qdp-odoo) [inline addons/account/account_move_line.py]: didn't you forget to cast counterpart as a list before doing counterpart[0:2] ??

File: addons/account/report/account_bank_reconciliation_report.py (hunk 1 above)
Comment (qdp-odoo) [inline addons/account/report/account_bank_reconciliation_report.py]: no, query_get() should only used when there is a need of using sql query directly... Here, it wouldn't add any value

File: addons/account/report/account_bank_reconciliation_report.py [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+                })
+                unrec_tot += -line.balance
+            lines.append(self.add_total_line(unrec_tot))
+
+        # Final
+        lines.append(self.add_title_line(_("Final balance"), start_amount + outstanding_plus_tot - outstanding_less_tot + unrec_tot))
+        return lines
+
+    @api.model
+    def _get_footnotes(self, type, target_id):
```
Comment (qdp-odoo) [inline addons/account/report/account_bank_reconciliation_report.py]: is it still needed?

File: addons/account/report/account_general_ledger.py [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+from openerp import models, fields, api, _
+
+
+class report_account_general_ledger(models.AbstractModel):
+    _name = "account.general.ledger"
+    _description = "General Ledger Report"
+
+    @api.model
+    def get_lines(self, context_id, line_id=None):
```
Comment (qdp-odoo) [inline addons/account/report/account_general_ledger.py]: the fact that get_lines() needs to be copied/pasted for each new custom reports makes me think that it should be defined at a higher level, in a class that all custom reports would inherit... and generic might inherit that part also,  don't you think?

**Comment by qdp-odoo** [comment]
it's merged... i've some remarks but i wrote them in a new task.

File: addons/account/report/account_general_ledger.py (hunk 8 above)
Comment (pvy-odoo) [inline addons/account/report/account_general_ledger.py]: What is added in the context depends on what filters are available in the configurator which is different from report to report hence the need to have this in the specific classes, I think.

**Pull request merged**